socket/test_socket.c

```diff
@@ -779,6 +793,236 @@ TEST(test_unix_socket, unix_transfer)
 }
 
 
+void unix_close_connected(int type)
+{
+	int fd[2];
+
+	if (socketpair(AF_UNIX, type, 0, fd) < 0)
+		FAIL("socketpair");
+
+	close(fd[0]);
+	close(fd[1]);
+}
+
+
+TEST(test_unix_socket, unix_close_connected)
+{
+	unsigned int i;
+
+	for (i = 0; i < CONNECTED_LOOP_CNT; ++i) {
+		unix_close_connected(SOCK_STREAM);
+		unix_close_connected(SOCK_DGRAM);
+		unix_close_connected(SOCK_SEQPACKET);
+	}
+}
+
+
+int got_epipe;
+
+void sighandler(int sig)
+{
+	got_epipe = 1;
+}
+
+
+void unix_send_after_close(int type, int epipe, int err)
+{
+	int fd[2];
+	ssize_t n;
+
+	signal(SIGPIPE, sighandler);
+
+	if (socketpair(AF_UNIX, type, 0, fd) < 0)
+		FAIL("socketpair");
+
+	close(fd[1]);
+
+	got_epipe = 0;
+	n = send(fd[0], data, sizeof(data), 0);
+	TEST_ASSERT(got_epipe == epipe);
+	TEST_ASSERT(n == -1);
+	TEST_ASSERT(errno == err);
+
+	got_epipe = 0;
+	n = send(fd[0], data, sizeof(data), 0);
+	TEST_ASSERT(got_epipe == epipe);
+	TEST_ASSERT(n == -1);
+	TEST_ASSERT(errno == err);
+
+	close(fd[0]);
```

Comment:
I don't know if we care or how to solve this, but won't this fd leak if some `TEST_ASSERT` fail?

Reply:
I assumed that leaking few descriptors is not a big deal - automatic tests run on a fresh system. I you really don't like it please add a new issue and I will fix later.